src/mouse/touch_handler.js

```diff
@@ -26,15 +26,16 @@ exports.addTouchListeners = function(el, editor) {
         var updateMenu = function() {
             var selected = editor.getCopyText();
             var hasUndo = editor.session.getUndoManager().hasUndo();
+            var commands = editor.commands.byName;
```

Comment:
small improvement:
```
var hasCommand = function(cmd) {
    var commands = editor.commands.byName;
    if(!commands[cmd]) return false;
    if(!editor.getReadOnly()) return false;
    return commands[cmd].readOnly;
};
```
this also disables commands when they are readonly

Reply:
do you want to remove non-readonly commands on readonly editor? Or for all editors?

Reply:
@whazor did you mean `if (!commands[cmd].readOnly && editor.getReadOnly())) return false` for the readonly part?
https://github.com/ajaxorg/ace/blob/master/src/commands/command_manager.js#L48C23-L48C39

Reply:
Yes, I think it would be nice to hide all commands that are unavailable. So in a readonly editor, only Select/Find/...

After looking at the command manager, maybe we could move the logic from 'exec' to a seperate method (`has`?) in the command manager and include both `readOnly` and `isAvailable`.

Reply:
New changes should make it, I think